.devcontainer/Dockerfile

```
@@ -1,31 +1,29 @@
-# Use the official Node.js 24 LTS image as base
-FROM node:24-bookworm
+# Multi-stage build for cross-platform compatibility
+# Using Ubuntu for better compatibility with devcontainer features
+FROM --platform=$BUILDPLATFORM mcr.microsoft.com/devcontainers/base:ubuntu
 
-# Install system dependencies and tools in a single layer
+# Install additional system dependencies that features might need
 RUN apt-get update && \
```

Hadolint (2.12.0)

[warning] Pin versions in apt get install. Instead of `apt-get install <package>` use `apt-get install <package>=<version>`

(DL3008)